File: LeetCode/UsingPython/554. Brick Wall.py
class Solution:
    def leastBricks(self, A: List[List[int]]) -> int:
        x = sum(A[0])
        arr = {}
        mx=0
        for i in range(len(A)):
            for j in range(len(A[i])-1):
                if j!=0:
                    A[i][j]+=A[i][j-1]
                if A[i][j]-1 not in arr:
                    arr[A[i][j]-1]=0
                arr[A[i][j]-1]+=1
                mx = max(mx,arr[A[i][j]-1])
        return len(A)-mx
        
        

File: LeetCode/C++/554. Brick Wall.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        unordered_map <int,int> mp;
        int crosses=0;
        int Max=0;
        for(int i=-0;i<wall.size();i++)
        {
            int sum=0;
            
            for(int j=0;j<wall[i].size()-1;j++)
            {
                sum+=wall[i][j];
                mp[sum]++;
                Max=max(Max,mp[sum]);
            }
            
        }
        crosses = wall.size()-Max;
        return crosses;
    }
};